src/Core/RulesPluginTrait.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Core\RulesPluginTrait.
+ */
+
+namespace Drupal\rules\Core;
+
+use Drupal\Component\Plugin\Exception\ContextException;
+
+/**
+ * Offers common methods for context plugin implementers.
+ */
+trait RulesPluginTrait {
```

Comment:
There are RulesPlugins which do not need this trait - thus the naming needs work. I'd suggest doing an ExecutablePluginTrait - as those methods stem from there.